submissions/f3d0t/friends-app/app.js

```diff
@@ -0,0 +1,166 @@
+const MAIN = document.querySelector(".main");
+const FILTERS = document.querySelector(".filters");
+const RESET_BUTTON = document.querySelector(".filters__button");
+const DAY_NIGHT = document.querySelector(".day_night");
+const FRIENDS_COUNT = 60;
+const REQUEST_LINK = `https://randomuser.me/api/?results=${FRIENDS_COUNT}`;
+
+const fetchData = async (requestLink) => {
+	try {
+		const response = await fetch(requestLink);
+		const { results } = await response.json();
+		return results;
+	} catch (error) {
+		alert("HTTP-Error: " + error);
+	}
+};
+
+const initApp = (friendsDataArray) => {
+	const friends = new FriendsList(friendsDataArray, MAIN);
+	friends.renderCards();
+	bindEventListeners(friends);
+};
+
+const bindEventListeners = (friends) => {
+	FILTERS.addEventListener("input", ({ target }) => {
+		if (target.name == "search") {
+			friends.filterBySearch(target.value);
+		}
+		if (target.name == "gender") {
+			friends.filterByGender(target.id);
+			FILTERS.elements.search.value = "";
+			FILTERS.elements.name.forEach((radioButton) => {
+				if (radioButton.checked) {
+					friends.sortByName(radioButton.id);
+				}
+			});
+			FILTERS.elements.age.forEach((radioButton) => {
+				if (radioButton.checked) {
+					friends.sortByAge(radioButton.id);
+				}
+			});//this two forEach callbacks is needed to re-sort new arrays of cards, filtered by gender
```

Comment:
How you can have items sorted by name and by gender in one time? 🤔

Reply:
Why not? 
Or you ask the sequence of actions?
My Gender filter really filters array - it is not sorting

Reply:
Sry, I meant sorted by name and by age.

Reply:
Some people can have the same age and in this case they will be sorted

Reply:
And if they not? :) What use case for this?

Imagine UX:

1. I am pressing sort by age - `0 > 99` sorted
1. I am pressing sort by name - 'a > z' sorted

And age sort, basically, ruined. But your application will show, that you sorting by age and by name. What the point?

Reply:
I will reset filtering by age after pressing name input for now.